contracts/Pool.sol

```diff
@@ -383,30 +390,59 @@ contract Pool is PoolFDT {
         emit Cooldown(msg.sender, uint256(0));
     }
 
+    /**
+        @dev   Update the effective deposit date based on how much new capital has been added.
+               If more capital is added, the depositDate moves closer to the current timestamp.
+        @dev   It emits a `DepositDateUpdated` event.
+        @param amt Total deposit amount.
+        @param who Address of user depositing.
+    */
+    function _updateDepositDate(uint256 amt, address who) internal {
+        uint256 prevDate = depositDate[who];
+        uint256 balance = balanceOf(who);
+
+        // prevDate + (now - prevDate) * (amt / (balance + amt))
+        // NOTE: prevDate = 0 implies balance = 0, and equation reduces to now
+        uint256 newDate = (balance + amt) > 0
+            ? prevDate.add(block.timestamp.sub(prevDate).mul(amt).div(balance + amt))
+            : prevDate;
+
+        depositDate[who] = newDate;
+        emit DepositDateUpdated(who, newDate);
+    }
+
     /**
         @dev Activates the cooldown period to withdraw. It can't be called if the user is not providing liquidity.
+        @dev It emits a `Cooldown` event.
     **/
     function intendToWithdraw() external {
-        PoolLib.intendToWithdraw(withdrawCooldown, balanceOf(msg.sender));
+        require(balanceOf(msg.sender) != uint256(0), "P:ZERO_BALANCE");
+        withdrawCooldown[msg.sender] = block.timestamp;
+        emit Cooldown(msg.sender, block.timestamp);
     }
 
     /**
         @dev Cancels an initiated withdrawal by resetting withdrawCooldown.
+        @dev It emits a `Cooldown` event.
     **/
     function cancelWithdraw() external {
-        PoolLib.cancelWithdraw(withdrawCooldown);
+        require(withdrawCooldown[msg.sender] != uint256(0), "P:NOT_WITHDRAWING");
+        withdrawCooldown[msg.sender] = uint256(0);
+        emit Cooldown(msg.sender, uint256(0));
```

Comment:
Moved over from `PoolLib`.

Reply:
Verified diff by copying into local develop branch in PoolLib, LGTM